backend/oasst_backend/models/journal.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime
+from typing import Optional
+from uuid import UUID, uuid1, uuid4
+
+import sqlalchemy as sa
+import sqlalchemy.dialects.postgresql as pg
+from sqlmodel import Field, SQLModel
+
+from .payload_column_type import PayloadContainer, payload_column_type
+
+
+def generate_time_uuid(node=None, clock_seq=None):
+    """Create a lexicographically sortable time ordered custom (non-standard) UUID by reordering the timestamp fields of a version 1 UUID."""
+    (time_low, time_mid, time_hi_version, clock_seq_hi_variant, clock_seq_low, node) = uuid1(node, clock_seq).fields
+    # reconstruct 60 bit timestamp, see version 1 uuid: https://www.rfc-editor.org/rfc/rfc4122
+    timestamp = (time_hi_version & 0xFFF) << 48 | (time_mid << 32) | time_low
+    version = time_hi_version >> 12
+    assert version == 1
+    a = timestamp >> 28  # bits 28-59
+    b = (timestamp >> 12) & 0xFFFF  # bits 12-27
+    c = timestamp & 0xFFF  # bits 0-11 (clear version bits)
+    clock_seq_hi_variant &= 0xF  # (clear variant bits)
+    return UUID(fields=(a, b, c, clock_seq_hi_variant, clock_seq_low, node), version=None)
```

Comment:
What is the point of using time ordered uuid?  Shouldn't we rely on `created_date` if we ever need to do something related to the time (e.g. weekly leaderboard).

Reply:
You may know the snowflake datatype (as used by twitter or discord). The idea here is the same: Time ordered unique ids that can be client-side generated on different machines. It will be used by integrator processes to track up to which point in time events have been processed.